packages/block-library/src/list-item/edit.js

```diff
@@ -8,30 +8,40 @@ import {
 } from '@wordpress/block-editor';
 import { __ } from '@wordpress/i18n';
 import { createBlock } from '@wordpress/blocks';
+import { useMergeRefs } from '@wordpress/compose';
 
-export default function ListItemEdit( {
-	name,
-	attributes,
-	setAttributes,
-	mergeBlocks,
-	onReplace,
-} ) {
+/**
+ * Internal dependencies
+ */
+import { useEnter } from './utils';
+
+export default function ListItemEdit( props ) {
+	const {
+		attributes,
+		setAttributes,
+		name,
+		onReplace,
+		mergeBlocks,
+		clientId,
+	} = props;
+	const { placeholder, content } = attributes;
 	const blockProps = useBlockProps();
 	const innerBlocksProps = useInnerBlocksProps( blockProps, {
 		allowedBlocks: [ 'core/list' ],
 	} );
-
+	const useEnterRef = useEnter( { content, clientId } );
 	return (
 		<li { ...innerBlocksProps }>
 			<RichText
+				ref={ useMergeRefs( [ useEnterRef ] ) }
```

Comment:
I added `useMergeRefs` here because it will be needed or other custom handling in the follow ups about split/merge at least.